aipl/ops/llm.py

```diff
@@ -107,20 +125,28 @@ def query_goose(aipl, v:str, **kwargs) -> str:
         temperature=0
     )
     params.update(**kwargs)
+    # TODO: GooseAI supports multiple prompt completions in parallel
     data = {'prompt': v, **params}
     r = requests.post(f'https://api.goose.ai/v1/engines/{model}/completions', headers=headers, json=data)
     j = r.json()
     if 'error' in j:
         raise AIPLException(f'''GooseAI returned an error: {j["error"]}''')
-    # TODO: check if it returns tokens used count
-    # cost = gooseai_pricing[model]['base'] + gooseai_pricing[model]['token']*used
-    # aipl.cost_usd += cost
-    # stderr(f'Used {used} tokens (estimate {len(v)//4} tokens).  Cost: ${cost:.04f}')
+    # TODO: GooseAI does not return number of tokens used, so we would need to use a tokenizer library to compute that separately
```

Comment:
I'm fine with a reasonable estimate of tokens, it doesn't have to be perfect, especially for a first pass.